.evergreen/mongosh_dl.py

```diff
@@ -35,55 +29,20 @@
 
 
 def _get_latest_version():
-    headers = {
-        "Accept": "application/vnd.github+json",
-        "X-GitHub-Api-Version": "2022-11-28",
-    }
-    if os.environ.get("GITHUB_TOKEN"):
-        headers["authorization"] = f"Bearer {os.environ['GITHUB_TOKEN']}"
-    url = "https://api.github.com/repos/mongodb-js/mongosh/releases"
-    req = urllib.request.Request(url, headers=headers)
-    try:
-        resp = urllib.request.urlopen(req, context=SSL_CONTEXT, timeout=30)
-        raw = resp.read().decode("utf-8")
-        data = json.loads(raw)
-        if not data:
-            LOGGER.debug("Releases (raw): %s", raw)
-            raise RuntimeError("List of releases is empty")
-        for item in data:
-            if item["prerelease"]:
-                continue
-            if item.get("draft"):
-                continue
-            return item["tag_name"].replace("v", "").strip()
-        raise RuntimeError(f"Could not find tag name in releases: {data}")
-    except Exception as e:
-        LOGGER.warning(
-            "Error fetching the latest version, falling back to using git tags"
-        )
-        LOGGER.warning(str(e))
-        return _get_latest_version_git()
-
-
-def _get_latest_version_git():
-    with tempfile.TemporaryDirectory() as td:
-        cmd = "git clone --depth 1 https://github.com/mongodb-js/mongosh.git"
-        subprocess.check_call(
-            shlex.split(cmd), cwd=td, stdout=subprocess.PIPE, stderr=subprocess.PIPE
-        )
-        cmd = "git fetch origin --depth=100 --tags"
-        path = os.path.join(td, "mongosh")
-        subprocess.check_call(
-            shlex.split(cmd), cwd=path, stdout=subprocess.PIPE, stderr=subprocess.PIPE
-        )
-        cmd = "git --no-pager tag"
-        output = subprocess.check_output(
-            shlex.split(cmd), cwd=path, stderr=subprocess.PIPE
-        )
-        for line in reversed(output.decode("utf-8").splitlines()):
-            if re.match(r"^v\d+\.\d+\.\d+$", line):
-                LOGGER.debug("Found version %s", line)
-                return line.replace("v", "").strip()
+    result = subprocess.run(
+        ["git", "ls-remote", "--tags", "https://github.com/mongodb-js/mongosh.git"],
+        stdout=subprocess.PIPE,
+        text=True,
+        check=False,
+    )
+    for line in reversed(result.stdout.splitlines()):
+        _, tag = line.split("refs/tags/")
+        match = re.match(r"^v(\d+\.\d+\.\d+)$", tag)
+        if match is not None:
+            tag = match.groups()[0]
+            LOGGER.debug("Found version %s", tag)
+            return tag
+    raise ValueError("Could not find tag!")
```

Comment:
Getting the latest mongosh version can be done best via https://downloads.mongodb.com/compass/mongosh.json rather than via github

See also the freshly merged https://github.com/mongodb-js/mongosh/commit/a7f68dc61bc9f80964a021f7ca0530f641e5fd4e (no Windows support yet)

Reply:
Thanks, updated